src/udev/udev-builtin-path_id.c

```diff
@@ -312,7 +312,7 @@ static struct udev_device *handle_scsi_iscsi(struct udev_device *parent, char **
         }
 
         format_lun_number(parent, &lun);
-        path_prepend(path, "ip-%s:%s-iscsi-%s-%s", addr, port, target, lun);
+        path_prepend(path, "ip-%s:%s-iscsi-%s-%s-%s", addr, port, target, lun, udev_device_get_sysname(parent));
```

Comment:
This does't look backwards compatible at all. You can add a new path, but the old one should be kept for backwards compatibility.

Please add some examples of how the new paths look like to the commit message. See #7411 for an excellent example.

Reply:
Thanks a lot for looking, I updated the comment.  I think (my opinion of course) that backwards compatibility in this case is not important.  The number of implementations that this potentially breaks is surely small, if existent at all (please correct me here if I am wrong or missing a large scale use-case).  The path_id for iscsi connected volumes is not unique to the path, and should be, making it incorrect.  It should be updated to be unique.

Reply:
The problem is that we don't know and cannot know. So if at all possible, we need to try to keep existing systems working after an upgrade. In particular, people might have just one iscsi device, in which case the old name is stable for them.

I know the old name doesn't make sense, but we can keep it by using both links (the one you added and the old one).

Reply:
This potential change should keep old links and add the new ones in as well, satisfying the backward compatibility issue.
~~~
diff --git a/rules/60-persistent-storage.rules b/rules/60-persistent-storage.rules
index 0de8cf3..f4d8260 100644
--- a/rules/60-persistent-storage.rules
+++ b/rules/60-persistent-storage.rules
@@ -73,7 +73,9 @@ KERNEL=="msblk[0-9]p[0-9]|mspblk[0-9]p[0-9]", ENV{ID_NAME}=="?*", ENV{ID_SERIAL}
 ENV{DEVTYPE}=="disk", DEVPATH!="*/virtual/*", IMPORT{builtin}="path_id"
 KERNEL=="mmcblk[0-9]boot[0-9]", ENV{DEVTYPE}=="disk", ENV{ID_PATH}=="?*", SYMLINK+="disk/by-path/$env{ID_PATH}-boot%n"
 KERNEL!="mmcblk[0-9]boot[0-9]", ENV{DEVTYPE}=="disk", ENV{ID_PATH}=="?*", SYMLINK+="disk/by-path/$env{ID_PATH}"
+ENV{DEVTYPE}=="disk", DEVPATH=="*/session[0-9]*/*", SYMLINK+="disk/by-path/$env{ID_PATH}-%P"
 ENV{DEVTYPE}=="partition", ENV{ID_PATH}=="?*", SYMLINK+="disk/by-path/$env{ID_PATH}-part%n"
+ENV{DEVTYPE}=="partition", DEVPATH=="*/session[0-9]*/*", SYMLINK+="disk/by-path/$env{ID_PATH}-part%n-%P"
 
 # legacy virtio-pci by-path links (deprecated)
 KERNEL=="vd*[!0-9]", ENV{ID_PATH}=="pci-*", SYMLINK+="disk/by-path/virtio-$env{ID_PATH}"
~~~

However, as of right now, the rule does not work because $parent and %P variable expansion seems to be broken (https://bugzilla.redhat.com/show_bug.cgi?id=1517374).  Just for an example, because I could test it, and because a picture is worth 1000 words, below shows the rule results, as it worked, changing $parent to $kernel.  Each unique path has a link and the old link remains as it was.
~~~
[root@localhost ~]# ls -lah /dev/disk/by-path | egrep 'sde|sdd'
lrwxrwxrwx. 1 root root   9 Nov 24 13:25 ip-192.168.124.12:3260-iscsi-iqn.2003-01.org.linux-iscsi.localhost.x8664:sn.f3618f5b946b-lun-0 -> ../../sde
lrwxrwxrwx. 1 root root   9 Nov 24 13:25 ip-192.168.124.12:3260-iscsi-iqn.2003-01.org.linux-iscsi.localhost.x8664:sn.f3618f5b946b-lun-0-sdd -> ../../sdd
lrwxrwxrwx. 1 root root   9 Nov 24 13:25 ip-192.168.124.12:3260-iscsi-iqn.2003-01.org.linux-iscsi.localhost.x8664:sn.f3618f5b946b-lun-0-sde -> ../../sde
~~~

Ideally, once $parent expansion is fixed, it would look like this:
~~~
lrwxrwxrwx. 1 root root   9 Nov 24 13:25 ip-192.168.124.12:3260-iscsi-iqn.2003-01.org.linux-iscsi.localhost.x8664:sn.f3618f5b946b-lun-0 -> ../../sde
lrwxrwxrwx. 1 root root   9 Nov 24 13:25 ip-192.168.124.12:3260-iscsi-iqn.2003-01.org.linux-iscsi.localhost.x8664:sn.f3618f5b946b-lun-0-20:0:0:0 -> ../../sdd
lrwxrwxrwx. 1 root root   9 Nov 24 13:25 ip-192.168.124.12:3260-iscsi-iqn.2003-01.org.linux-iscsi.localhost.x8664:sn.f3618f5b946b-lun-0-21:0:0:0 -> ../../sde
~~~

If you guys find this more desirable than the initial commit I can certainly edit it.  Or if you have more info/issues I will help if I can.

Reply:
I think you are looking for ```$id``` or ```%b``` instead of ```$parent```
https://bugzilla.redhat.com/show_bug.cgi?id=1517377#c7

Reply:
Thanks Harald et al. I have supplied a second commit, keeping things backward compatible.